salt/iaticdfdbackend.sls

```diff
@@ -0,0 +1,260 @@
+
+include:
+  - core
+  - apache
+  - uwsgi
+  - letsencrypt
+
+##################################################################### Normal User
+
+{% set user = 'iaticdfdbackend' %}
+{{ createuser(user) }}
+
+##################################################################### Logging User with special access
+{% set user_logs = 'iaticdfdbackend_logs' %}
+{{ createuser(user_logs) }}
+
+add_{{ user_logs }}_to_group_adm:
+  group.present:
+    - name: adm
+    - addusers:
+      - {{ user_logs }}
+
+##################################################################### Ubuntu Dependencies
+
+iaticdfdbackend-deps:
+    apache_module.enabled:
+      - name: proxy proxy_uwsgi rewrite
+      - watch_in:
+        - service: apache2
+    pkg.installed:
+      - pkgs:
+        - libapache2-mod-proxy-uwsgi
+        - python3-pip
+        - python3-virtualenv
+        - uwsgi-plugin-python3
+        #- redis-server
+        - postgresql-12
+        - libpq-dev
+        - gcc
+        - make
+        - libxml2-dev
+        - libxslt1-dev
+        - libevent-dev
+        - python3-dev
+      - watch_in:
+        - service: apache2
+        - service: uwsgi
+
+##################################################################### Continuous Deployment needs sudo
+
+/etc/sudoers.d/{{ user }}-global:
+  file.managed:
+    - source: salt://iaticdfdbackend/sudoers-global
+    - template: jinja
+    - context:
+        user: {{ user }}
+
+##################################################################### Macro to install app
+
+{% macro iaticdfdbackend(name, giturl, branch, codedir, webserverdir, user, uwsgi_port, https, servername, postgres_name, postgres_user, postgres_password , uwsgi_as_limit, uwsgi_harakiri, uwsgi_workers, uwsgi_max_requests, uwsgi_reload_on_as, sentry_dsn, sentry_traces_sample_rate) %}
+
+###################### Code folder & virtual env & Python Libs & more
+
+{{ giturl }}{{ codedir }}:
+  git.latest:
+    - name: {{ giturl }}
+    - rev: {{ branch }}
+    - target: {{ codedir }}
+    - user: {{ user }}
+    - force_fetch: True
+    - force_reset: True
+    - require:
+      - pkg: git
+    - watch_in:
+      - service: uwsgi
+
+{{ codedir }}.ve/:
+  virtualenv.managed:
+    - python: /usr/bin/python3
+    - user: {{ user }}
+    - system_site_packages: False
+    - require:
+      - pkg: iaticdfdbackend-deps
+      - git: {{ giturl }}{{ codedir }}
+    - watch_in:
+      - service: apache2
+
+# Fix permissions in virtual env
+{{ codedir }}fix-ve-permissions:
+  cmd.run:
+    - name: chown -R {{ user }}:{{ user }} .ve
+    - user: root
+    - cwd: {{ codedir }}
+    - require:
+      - virtualenv: {{ codedir }}.ve/
+
+# This should ideally be in virtualenv.managed but we get an error if we do that
+{{ codedir }}install-python-packages:
+  cmd.run:
+    - name: . .ve/bin/activate; pip install -r requirements.txt
+    - user: {{ user }}
+    - cwd: {{ codedir }}
+    - require:
+      - virtualenv: {{ codedir }}.ve/
+
+# Remove file
```

Comment:
Why do we need to remove this file? And if so, does it need to be in the Virtenv to do it? You can just use https://docs.saltproject.io/en/latest/ref/states/all/salt.states.file.html#salt.states.file.absent

Can the comment say why it needs to be removed?

Reply:
Comment explaining added to salt file